### PR TITLE
Vector phasing pipeline spec

### PR DESCRIPTION
**Pull request by alimanfoo**
This PR has an initial draft of a phasing pipeline for vector using WhatsHap and SHAPEIT4.

### CODE REVIEW

**Comment by alimanfoo** [comment]
Hi @hardingnj, @jonbrenas, could we meet up to talk this one through? I put an initial sketch together based on reading the docs of whatshap and shapeit4 but may have missed things, would be good to get your thoughts.

**Comment by alimanfoo** [comment]
FYI @kbergin, @kemprg, @gbggrant, starting work here towards a phasing pipeline spec. I need to go through this with @hardingnj and @jonbrenas who have some experience of this, so probably not worth looking at too closely yet, but any questions/comments welcome of course.

File: docs/specs/phasing-vector.md [hunk 1]
```diff
@@ -0,0 +1,254 @@
+# Phasing (vector) pipeline specification
```
Comment (kbergin) [inline docs/specs/phasing-vector.md:1]: Perhaps "Mosquito Vector Genotype Phasing Pipeline" ?

File: docs/specs/phasing-vector.md [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+phasing and statistical (population) phasing to improve phasing
+accuracy.
+
+
+## Inputs
+
+* A manifest specifying a set of samples to be phased.
+
+* For each sample, a BAM file containing analysis-ready sequence read
+  alignments, produced by the short read alignment pipeline.
```
Comment (kbergin) [inline docs/specs/phasing-vector.md:18]:
```suggestion
  alignments, produced by the [short read alignment pipeline](https://github.com/malariagen/pipelines/blob/master/docs/specs/short-read-alignment-vector.md).
```

File: docs/specs/phasing-vector.md [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+
+* For each sample, a VCF file or zipped zarr file containing unphased
```
Comment (kbergin) [inline docs/specs/phasing-vector.md:20]: Do we have a spec for that pipeline yet?

File: docs/specs/phasing-vector.md [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+
+## Implementation notes
+
+@@TODO something about chunking and parallelisation
+
+* SHAPEIT4 has optimisations to use AVX2 instruction set, and so
+  compiling and running the program on CPUs supporting AVX2 is
+  preferable. See also [SHAPEIT4 installation
+  notes](https://odelaneau.github.io/shapeit4/#installation).
+
```
Comment (alimanfoo) [inline docs/specs/phasing-vector.md:250]: Open question: how to handle missing genotype calls? Reasonable to impute, or should be left as missing? If want to leave as missing, does this mean we need to post-process data, if SHAPEIT4 always performs imputation?

File: docs/specs/phasing-vector.md (hunk 1 above)
Comment (alimanfoo) [reply docs/specs/phasing-vector.md]: Renamed.

File: docs/specs/phasing-vector.md (hunk 2 above)
Comment (alimanfoo) [reply docs/specs/phasing-vector.md]: Added link.

File: docs/specs/phasing-vector.md (hunk 3 above)
Comment (alimanfoo) [reply docs/specs/phasing-vector.md:21]: Yep, #20, added link.

**Comment by alimanfoo** [comment]
Propose to merge this PR as-is shortly if no objections, noting that the document is a draft and we may need to revisit later during implementation if anything comes up then.

**Pull request merged**